pinc/smoothread.inc

```diff
@@ -28,31 +28,25 @@ include_once($relPath.'user_project_info.inc'); // notify_project_event_subscrib
 *
 * Remarks:
 *
-* Currently only a check for existence of record and field being <> 0 is done. The latter
+* Currently only a check for existence of record and field being > 0 is done. The latter
 * would allow the extension for a timestamp and the resetting to 0 if revoked in order
 * to still show users that have once committed but then revoked that commitment.
 *
 ****************************************************************************************/
 
 function sr_user_is_committed($projectid, $username)
 {
-    $res = mysqli_query(DPDatabase::get_connection(), "
-        SELECT *
+    $sql = sprintf("
+        SELECT count(*)
         FROM smoothread
-        WHERE projectid = '$projectid' 
-        AND user = '$username'
-        ") or die(DPDatabase::log_error());
-
-    $row = mysqli_fetch_assoc($res);
-    if(!$row)
-        return false;
-
-    if ($row['committed'] <> 0)
-    {
-        return true;
-    }
-
-    return false;
+        WHERE projectid = '%s'
+            AND user = '%s'
+            AND committed > 0
```

Comment:
it's never negative right? Before we had <> 0 and now just > 0

Reply:
Fair question. The field is a `tinyint` and right now we only store 1s in it (we delete rows instead of setting it to 0) so `> 0` is valid for how we're currently using it.